vaadin-grid-flow-parent/vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/Grid.java

```diff
@@ -2401,6 +2474,74 @@ public FooterRow appendFooterRow() {
         return insertColumnLayer(getLastFooterLayerIndex() + 1).asFooterRow();
     }
 
+    /**
+     * Removes the footer row from the grid.
+     *
+     * @see #removeAllFooterRows()
+     * @param footerRow
+     *            the footer row to remove
+     * @throws NoSuchElementException
+     *             if the footer row cannot be found
+     */
+    public void removeFooterRow(FooterRow footerRow) {
+        Objects.requireNonNull(footerRow);
+        if (!getFooterRows().contains(footerRow)) {
+            throw new NoSuchElementException(
+                    "Footer to remove cannot be found.");
+        }
+        if (getColumnLayers().get(0).equals(footerRow.layer)) {
+            // Bottom layer needs special handling and content manipulation
+            // because it cannot be removed.
+            removeFooterInBottomLayer(footerRow);
+        } else {
+            removeColumnLayer(footerRow.layer);
+        }
+    }
+
+    private void removeFooterInBottomLayer(FooterRow footerRow) {
+        List<FooterRow> footerRows = getFooterRows();
+        if (footerRows.size() == 1) {
+            // There is no other footer row to move content from. Therefore, the
+            // layer is cleared.
+            footerRow.getCells()
+                    .forEach(footerCell -> footerCell.setText(null));
+            footerRow.layer.setFooterRow(null);
+            footerRow.layer = null;
+            return;
+        }
+        // Move content from next footer row and remove that layer
+        FooterRow nextFooterRow = footerRows
+                .get(footerRows.indexOf(footerRow) + 1);
+        ColumnLayer layerToRemove = nextFooterRow.layer;
+        moveRowContent(nextFooterRow, footerRow);
+        footerRow.layer.setFooterRow(nextFooterRow);
+        layerToRemove.setFooterRow(null);
+        removeColumnLayer(layerToRemove);
```

Comment:
Since we're not necessarily removing the given row in case it's the bottom layer you might experience some unexpected side effects:

```java
grid.appendFooterRow();
grid.appendFooterRow();

var row = grid.getFooterRows().get(0);
grid.removeFooterRow(row);
row.getCell(grid.getColumns().get(0)).setText("I'm not really removed");
```

This might be an edge-case issue though

Reply:
No cell should be open to manipulation through a removed cell. So I added a [method](https://github.com/vaadin/flow-components/pull/5990/files#diff-38b24105752fa2cd7421e5853010a022ee5add4320638b46ce7ffaeb89d80b5dR2537-R2541) to clear the row content. It removes the layer reference and clears the cell list. I added this logic to all spots where a row in bottom layer is removed (default header, non-default header, only footer, multiple footers). I also added [several tests](https://github.com/vaadin/flow-components/pull/5990/files#diff-f1ce143dd41c8004714d3340d368282868b6da90bccf6d3faacbe8d3304ad8c5R1319-R1379) to ensure that no cell can be modified via a removed row.